### DISCUSSION TITLE
Is there method to call native function or python function from javascript in web?

### DISCUSSION THREAD

**Discussion by zeroxer** [beeware/toga#1843] [Q&A]
### What is the problem or limitation you are having?

- Hope to call python code from javascript.

### Describe the solution you'd like

- send message from javascript to python.
- send response from python to javascript. 

### Describe alternatives you've considered

Add native interface to glue python and javascript.

**Comment by freakboy3742** [comment]
Yes, this is entirely possible; The Toga web backend is based on [Pyscript](https://pyscript.net), which is itself based on [Pyodide](https://pyodide.org/). Any of the techniques described by those projects will work inside Toga's web interface - with the caveat that the code is platform specific, so it won't work on other platforms. 

The short version: `import js` will give you the same namespace you have on a web page. For example `js.document.get_element_by_id('foo')` will retrieve a DOM element with an ID of `foo`; if your Python code contains `element = js.document.create_element()` to create a new element, you can use `element.onclick = myfunc` (where `my_func` is a Python function) to install a click handler.
